controllers/bootstrap/init.go

```diff
@@ -1374,11 +1376,48 @@ func (b *Bootstrap) updateApprovalMode() error {
 		return err
 	}
 
-	if err = b.UpdateCsOpApproval(); err != nil {
+	if err = b.UpdateOpApproval(constant.IBMCSPackage); err != nil {
 		klog.Errorf("Failed to update common service operator subscription: %v", err)
 		return err
 	}
 
+	if err = b.UpdateOpApproval(constant.IBMODLMSPackage); err != nil {
+		klog.Errorf("Failed to update operand-deployment-lifecycle-manager-app subscription: %v", err)
+		return err
+	}
+
+	if err = b.UpdateOpApproval(constant.IBMNSSPackage); err != nil {
+		klog.Errorf("Failed to update namespace-scope-operator subscription: %v", err)
+		return err
+	}
+
+	if err = b.UpdateOpApproval(constant.ICPOperator); err != nil {
+		if !errors.IsNotFound(err) {
+			klog.Errorf("Failed to update %s subscription: %v", constant.ICPOperator, err)
+			return err
+		}
+		klog.Infof("%s not installed, skipping updating approval strategy", constant.ICPOperator)
+
+	}
+
+	if err = b.UpdateOpApproval(constant.ICPPICOperator); err != nil {
+		if !errors.IsNotFound(err) {
+			klog.Errorf("Failed to update %s subscription: %v", constant.ICPPICOperator, err)
+			return err
+		}
+		klog.V(2).Infof("%s not installed, skipping", constant.ICPPICOperator)
+
+	}
+
+	if err = b.UpdateOpApproval(constant.ICPPKOperator); err != nil {
+		if !errors.IsNotFound(err) {
+			klog.Errorf("Failed to update %s subscription: %v", constant.ICPPKOperator, err)
+			return err
+		}
+		klog.V(2).Infof("%s not installed, skipping", constant.ICPPKOperator)
+
+	}
```

Comment:
I think we need to move this part to the place where crossplane is installed.
Crossplane is installed after we initialize the resources. Therefore, if we update subscription of crossplane here, it will usually skip the update because it has not been installed yet.
https://github.com/IBM/ibm-common-service-operator/blob/master/controllers/commonservice_controller.go#L147
https://github.com/IBM/ibm-common-service-operator/blob/master/controllers/commonservice_controller.go#L234

Reply:
Yeah, I agree, will change it.